sql/analyzer/resolve_column_defaults.go

```diff
@@ -0,0 +1,480 @@
+	"st_isclosed": {},
+	"st_isempty": {},
+	"st_issimple": {},
+	"st_isvalid": {},
+	"st_latfromgeohash": {},
+	"st_latitude": {},
+	"st_length": {},
+	"st_linefromtext": {},
+	"st_linestringfromtext": {},
+	"st_linefromwkb": {},
+	"st_linestringfromwkb": {},
+	"st_longfromgeohash": {},
+	"st_longitude": {},
+	"st_makeenvelope": {},
+	"st_mlinefromtext": {},
+	"st_multilinestringfromtext": {},
+	"st_mlinefromwkb": {},
+	"st_multilinestringfromwkb": {},
+	"st_mpointfromtext": {},
+	"st_multipointfromtext": {},
+	"st_mpointfromwkb": {},
+	"st_multipointfromwkb": {},
+	"st_mpolyfromtext": {},
+	"st_multipolygonfromtext": {},
+	"st_mpolyfromwkb": {},
+	"st_multipolygonfromwkb": {},
+	"st_numgeometries": {},
+	"st_numinteriorring": {},
+	"st_numinteriorrings": {},
+	"st_numpoints": {},
+	"st_overlaps": {},
+	"st_pointfromgeohash": {},
+	"st_pointfromtext": {},
+	"st_pointfromwkb": {},
+	"st_pointn": {},
+	"st_polyfromtext": {},
+	"st_polygonfromtext": {},
+	"st_polyfromwkb": {},
+	"st_polygonfromwkb": {},
+	"st_simplify": {},
+	"st_srid": {},
+	"st_startpoint": {},
+	"st_swapxy": {},
+	"st_symdifference": {},
+	"st_touches": {},
+	"st_transform": {},
+	"st_union": {},
+	"st_validate": {},
+	"st_within": {},
+	"st_x": {},
+	"st_y": {},
+	"statement_digest": {},
+	"statement_digest_text": {},
+	"std": {},
+	"stddev": {},
+	"stddev_pop": {},
+	"stddev_samp": {},
+	"str_to_date": {},
+	"strcmp": {},
+	"subdate": {},
+	"substr": {},
+	"substring": {},
+	"substring_index": {},
+	"subtime": {},
+	"sum": {},
+	"sysdate": {},
+	"system_user": {},
+	"tan": {},
+	"time": {},
+	"time_format": {},
+	"time_to_sec": {},
+	"timediff": {},
+	"timestamp": {},
+	"timestampadd": {},
+	"timestampdiff": {},
+	"to_base64": {},
+	"to_days": {},
+	"to_seconds": {},
+	"trim": {},
+	"truncate": {},
+	"ucase": {},
+	"uncompress": {},
+	"uncompressed_length": {},
+	"unhex": {},
+	"unix_timestamp": {},
+	"updatexml": {},
+	"upper": {},
+	"user": {},
+	"utc_date": {},
+	"utc_time": {},
+	"utc_timestamp": {},
+	"uuid": {},
+	"uuid_short": {},
+	"uuid_to_bin": {},
+	"validate_password_strength": {},
+	"values": {},
+	"var_pop": {},
+	"var_samp": {},
+	"variance": {},
+	"version": {},
+	"wait_for_executed_gtid_set": {},
+	"wait_until_sql_thread_after_gtids": {},
+	"week": {},
+	"weekday": {},
+	"weekofyear": {},
+	"weight_string": {},
+	"year": {},
+	"yearweek": {},
+}
+
+func resolveColumnDefaults(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (sql.Node, error) {
+	span, _ := ctx.Span("resolveColumnDefaults")
+	defer span.Finish()
+
+	return plan.TransformUp(n, func(n sql.Node) (sql.Node, error) {
+		switch node := n.(type) {
+		case sql.SchemaModifiable:
+			sch := node.Schema()
+			var columns = make(map[string]indexedCol)
+			for i, col := range sch {
+				columns[strings.ToLower(col.Name)] = indexedCol{col, i}
+			}
+			newSch := make(sql.Schema, len(sch))
+
+			for i, col := range sch {
+				newCol := *col
+				if col.Default == nil || col.Default.Resolved() {
```

Comment:
Pretty sure this expression will always be true if the defaults are valid, which means none of the check code below will ever run. Needs tests.